Shared/Extensions/URLExtensions.swift

```diff
@@ -316,6 +316,12 @@ extension URL {
         }
         return urls[0] == urls[1]
     }
+    
+    public var isFxHomeUrl: Bool {
+        if self.absoluteString.hasPrefix("internal://") { return true }
+        return false
+    }
```

Comment:
nit: I think this can be simplified like that? Without `self.` too
```suggestion
    public var isFxHomeUrl: Bool {
        return absoluteString.hasPrefix("internal://")
    }
```
And since this is an extension on URL, we could directly use that to clean up `isURLStartingPage` and `isFxHomeTab` in Tab.swift and a couple other places. I guess it should have been an extension from the get go since it's duplicated in 5 places.